JumpscaleCore/core/BASECLASSES/JSConfigsBCDB.py

```diff
@@ -188,20 +190,10 @@ def reset(self, recursive=None):
             except Exception as e:
                 j.shell()
 
-        mother_id = self._mother_id_get()
-        # TODO: in fact this should work without having to do this, we put it in to get to some more tests, but if deletes work well it should work without (JO)
-        if not mother_id:
-            # means we can remove all objects of the url (index)
+        _, has_mother = self._mother_id_get()
+        if not has_mother:
             self._model.index.destroy()
-        else:
-            # if we did not specify and there is a mother_id for sure we can do the recursive behaviour
-            if recursive == None:
-                recursive = True
-
-        if recursive:
-            self._children_delete(recursive=recursive)
```

Comment:
This step is not needed. The children are already deleted at the beginning of the function.